gips/0081-indexing-payments.md

```diff
@@ -0,0 +1,165 @@
+
+The Indexer stack will be updated to be able to accept indexing agreements from the dipper when the price is above the price set by the Indexer, and up to a maximum amount of subgraphs also set by the Indexer.
+
+Pricing between Indexer and gateway in the MVP will be based on the "amount of work done" proxy detailed above.
+
+The dipper will initially set up agreements with 3 Indexers for each desired subgraph, as we estimate this is a reasonable number to achieve 99.9% uptime in most subgraphs.
+
+Once the smart contracts proposed in this GIP are deployed, we anticipate adapting the indexer and gateway components accordingly.
+
+The MVP has different trust implications than what this proposal has described so far. In particular these are:
+
+- Indexers do the work upfront and trust that the gateway will pay them accordingly after the fact, since the gateway issues payment vouchers upon receiving PoIs and not before.
+- Gateways trust that indexers are reporting the correct amount of work done, since this information is not posted on-chain and thus it's non-slashable. This is partially mitigated by the fact that the gateway can (theoretically) compare the reported work done among the pool of indexers selected and thus can weed out bad actors, but it's nevertheless a weaker mechanism than slashing.
+
+## Detailed Specification
+
+TODO
+
+## Dependencies
+
+This GIP depends on Horizon being fully functional.
+
+## Rationale and Alternatives
+
+When presenting this proposal, we often hear suggestions for alternative, and possibly simpler, approaches to payments, for instance:
+
+- Could the Gateway just pre-pay a fixed amount per subgraph?
+- Could Indexers just charge a specific (Indexer-determined) amount for each subgraph in arrears?
+
+However, the design of The Graph protocol is, and must be, adversarial. When designing incentives, participants are generally considered rational, but the protocol must be safe against byzantine participants that may derive profit from malicious actions in ways that cannot be predicted within the protocol. The typical cryptoeconomic approach for this is to establish a cost-of-corruption, so that malicious actors are punished and participants can quantify the risk in each interaction.
+
+In the simple examples mentioned above, there are a few potential attack vectors to consider: the Gateway could refuse payment after the fact, or the Indexer could take the money and run, or the Indexer could try to charge a ridiculous amount after the Gateway has promised payment.
+
+Considering this, the protocol is designed to minimize the need for trust, in two directions:
+
+- Gateways should not need to trust that Indexers will do their job or provide correct data. Otherwise, Gateways would only query trusted Indexers (e.g. Indexers with a reputation or direct relationship with the Gateway, which could lead to centralization or a winner-take-all Indexer.)
```

Comment:
```suggestion
- Gateways should not need to trust that Indexers will do their job or provide correct data. Otherwise, Gateways would only query trusted Indexers (e.g. Indexers with a reputation or direct relationship with the Gateway, which could lead to centralization or a winner-take-all Indexer).
```